httpclient/HTTPClient.cpp

```diff
@@ -111,15 +129,12 @@ void HTTPClient::SetUserAgent(const std::string &useragent)
 	m_sUserAgent = useragent;
 }
 
-void HTTPClient::LogError(void *curlobj)
+void HTTPClient::LogError(const long response_code)
```

Comment:
Could you undo this change? Now you create a lot more code that did not have to be created. This function was OK

Reply:
Actually I tried to reduce as much as possible code, which is why I changed this part to use just the already retrieved response code (since we need this for event processing as well). Now we only use CURLINFO_RESPONSE_CODE once on each call. Could also extend the LogError function to process it for returning the code, but IMO that isn't the correct place to do it. 

I do believe the return code is very important, so one way or the other we need to include it somewhere (and I decided to put it at the end of the header return vector, to use as little as possible return vars)

Reply:
If a HTTP request (get/post/binary) does NOT return successfully , and only if (so there is a problem) that log function is called.
If it success, it returns TRUE... you can use this function in your dzVents to check if your call succeeded.
Next to this, for sure users will try the call in the browser and know the mime type
Next, when it should be json (call a function IsJSON for example) and you expect a temperature field, check for this field, else the call returned unwanted result.
No need for headers or mime type, just use the already returned Boolean to check if the call succeeded.
As for maybe in the feature... yes maybe in the feature

Reply:
It is asynchronous so you cannot wait for the return value. OpenURL only fills the commandArray and long after the script is finished the results will be fetched by Domoticz and this will again trigger dzVents. With the callback trigger dzVents will see which script needs to be called with the response data.

Reply:
So if I understand it correctly, you only want to trigger dzVents with the response data only if there was a success? And if that's the case the raw response body is handed over to  dzVents and dzVents will call the callback-script with this raw data and the scripter needs to know what the mime-type is to be expected and it to json when needed himself?

Reply:
Yes agree, could only parse / return status code if != 200, but that's a minor change (moving a few lines)... So you would really only rely return status true / false for processing in scripts? Makes things much less flexible unfortunately.

Reply:
Yes.. it's a bit of a bummer indeed.